python-package/README.rst

```diff
@@ -84,7 +84,7 @@ Build GPU Version
 
 .. code:: sh
 
-    pip install lightgbm --install-option=--gpu
+    pip install lightgbm --config-setting=cmake.define.USE_GPU=ON
```

Comment:
Pip exposes `--config-settings` option: https://pip.pypa.io/en/stable/cli/pip_install/#cmdoption-C. Is `--config-setting` coming from somewhere else?

Reply:
Oh wow, thank you! No you're absolutely right, this is a mistake on my part.

I was mixing two things... `pip` has `--config-settings` (with an `s`), and `build` has `--config-setting` (no `s`) ([code link](https://github.com/pypa/build/blob/e0c59cc4e17e63cfe9dbd4f366fae6d51b073282/src/build/__main__.py#L331-L339)).

I was following from https://github.com/scikit-build/scikit-build-core/blob/a145985892e4c904f4e4f089f987b4fef491543d/README.md?plain=1#L112-L113 and didn't read closely enough.

Just pushed https://github.com/microsoft/LightGBM/pull/5759/commits/41badff4121b6003c95e8dbc8e967348e1820e96 fixing that.